.github/workflows/python-push.yml

```diff
@@ -78,28 +78,45 @@ jobs:
 
   deploy:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    environment:
+      name: release
+      url: https://pypi.org/p/compliance-trestle-fedramp
     needs: build
     if: github.ref == 'refs/heads/main' && github.repository == 'oscal-compass/compliance-trestle-fedramp'
     steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: true
-        fetch-depth: 0
-        token: ${{ secrets.ADMIN_PAT }} 
-    - name: Set up Python 3.9
-      uses: actions/setup-python@v2
-      with:
-        python-version: 3.9
-    - name: Install build tools
-      run: |
-        make develop
-    - name: Create release
-      shell: bash
-      env:
-        PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
-        GH_TOKEN: ${{ secrets.ADMIN_PAT }}
-      run: |
-        make release
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+          token: ${{ secrets.ADMIN_PAT }} 
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+      - name: Install build tools
+        run: |
+          make develop
+
+      # This action uses Python Semantic Release v8
+      - name: Python Semantic Release
+        id: release
+        uses: python-semantic-release/python-semantic-release@v9.8.0
+        with:
+          github_token: ${{ secrets.ADMIN_PAT }}
+
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        # NOTE: DO NOT wrap the conditional in ${{ }} as it will always evaluate to true.
+        # See https://github.com/actions/runner/issues/1173
+        if: steps.release.outputs.released == 'true'
+
+      - name: Publish package distributions to GitHub Releases
+        uses: python-semantic-release/upload-to-gh-release@v9.8.0
+        if: steps.release.outputs.released == 'true'
+        with:
+          github_token: ${{ secrets.ADMIN_PAT }}
 
   merge-main-to-develop:
```

Comment:
does merge main to develop also needs to use checkout@v4 similar to trestle?
  steps:
    - uses: actions/checkout@v2

Reply:
Thank you. I will update this.